[XLA:GPU] Add Intel GPU specific tags for xla and sysl_status component

kanvi-nervana opened the pull request:
co-authors: @agramesh1 @bhavani-subramanian @mraunak 
This PR enables us to add Intel GPU tests, without breaking ROCm/cuda testing.

1. Adds oneapi/intel_gpu equivalents of rocm/cuda tags to XLA and adds them to rocm/cuda build/CI scripts so they will skip Intel GPU specific tests.
2. Adds the sycl_status component and test.
3. As part of this change, the use of sycl is replaced with oneapi when referring to tools and backend.

kanvi-nervana commented:
@penpornk Please help review this PR. Thanks!

beckerhe commented on xla/stream_executor/sycl/sycl_status.h:
```diff
@@ -0,0 +1,51 @@
+#define XLA_STREAM_EXECUTOR_SYCL_SYCL_STATUS_H_
+
+#include <string>
+#include <sycl/sycl.hpp>
```
I see this includes sycl-specific headers. How do we get this to build on our CI? Do you already have some Bazel build rules that make Bazel build SYCL, or download prebuilt libraries and headers?

(I'm reluctant to accept code that we can't build because it makes refactoring so much harder.)

kanvi-nervana replied:
@beckerhe The change to add clang build support as well as hermetic build is being worked on and should be ready soon. We can put this on hold until then.

dimitar-asenov replied:
@kanvi-nervana, Could you please mark this PR as draft, until all prerequisites are ready and submitted? Once this PR is ready to be reviewed again, please remove the draft status and ping us here.

kanvi-nervana replied:
@dimitar-asenov I have updated the PR since the pre-requisites have been merged. The PR is ready for review. Thanks!

beckerhe commented:
Hey, can you define a GitHub Actions job for this build config?

It should use the same machine pool as the X86 CPU build and should only run SYCL-specific tests. The build config needs to be defined in https://github.com/openxla/xla/blob/main/build_tools/ci/build.py and the job needs to be registered in https://github.com/openxla/xla/blob/main/.github/workflows/ci.yml

kanvi-nervana commented:
> Hey, can you define a GitHub Actions job for this build config?
> 
> It should use the same machine pool as the X86 CPU build and should only run SYCL-specific tests. The build config needs to be defined in https://github.com/openxla/xla/blob/main/build_tools/ci/build.py and the job needs to be registered in https://github.com/openxla/xla/blob/main/.github/workflows/ci.yml

@beckerhe We are working on it.

kanvi-nervana commented:
> > Hey, can you define a GitHub Actions job for this build config?
> > It should use the same machine pool as the X86 CPU build and should only run SYCL-specific tests. The build config needs to be defined in https://github.com/openxla/xla/blob/main/build_tools/ci/build.py and the job needs to be registered in https://github.com/openxla/xla/blob/main/.github/workflows/ci.yml
> 
> @beckerhe We are working on it.

@beckerhe Added the requested changes, please review the PR. Thanks!

beckerhe commented on build_tools/ci/build.py:
```diff
@@ -305,7 +313,30 @@ def nvidia_gpu_build_with_compute_capability(
+    configs=("sycl", "sycl_hermetic"),
+    # This build of oneAPI backend runs on X86 host without an Intel GPU, so 
+    # build command is passed here instead of test.
+    subcommand="build",
```
As mentioned above - if we filter test targets that require a GPU then we can set this to `test` and still run all the tests that don't require a GPU.

beckerhe commented on xla/stream_executor/sycl/BUILD:
```diff
@@ -37,6 +37,10 @@ cc_library(
     srcs = ["sycl_platform.cc"],
     hdrs = ["sycl_platform.h"],
     visibility = ["//visibility:public"],
+    tags = [
+        "gpu",
+        "oneapi-only",
+    ],
```
Since this does not depend on anything GPU- or ONEAPI-specific there is no need to tag this as `gpu` or `oneapi-only`.

mraunak commented:
Hi @beckerhe we have addressed the comments.

kanvi-nervana commented:
> I've added some minor comments but in general this looks good to me now.

@beckerhe I have addressed your comments.

kanvi-nervana commented:
> > > I've added some minor comments but in general this looks good to me now.
> > 
> > 
> > @beckerhe I have addressed your comments.
> 
> Thank you. Can you rebase the PR. There appear to be some unrelated errors in the presubmits.

@beckerhe I have rebased the PR
